### ISSUE TITLE
Random test failure `routing::archival_node`

### ISSUE THREAD

**Issue by pmnoxx** [near/nearcore#5389]
We have seen a random test failure for `routing::archival_node`

https://buildkite.com/nearprotocol/nearcore/builds/7640#d7e5d254-c79d-443b-9a07-58df975f5130

Plan:
- [ ] Add debug logs to `runner.rs` https://github.com/near/nearcore/pull/5390
- [ ] Add fix https://github.com/near/nearcore/pull/5639

**Comment by pmnoxx** [comment]
```


[2021-11-19T18:02:42Z] test routing::archival_node ... FAILED
--
  | [2021-11-19T18:02:42Z]
  | [2021-11-19T18:02:42Z] failures:
  | [2021-11-19T18:02:42Z]
  | [2021-11-19T18:02:42Z] ---- routing::archival_node stdout ----
  | [2021-11-19T18:02:42Z] BBBB Slept 15001; max_wait_ms 15000
  | [2021-11-19T18:02:42Z] thread 'routing::archival_node' panicked at 'Timed out waiting for the condition', chain/network/src/test_utils.rs:159:21
  | [2021-11-19T18:02:42Z] stack backtrace:
  | [2021-11-19T18:02:42Z]    0: rust_begin_unwind
  | [2021-11-19T18:02:42Z]              at /rustc/09c42c45858d5f3aedfa670698275303a3d19afa/library/std/src/panicking.rs:517:5
  | [2021-11-19T18:02:42Z]    1: core::panicking::panic_fmt
  | [2021-11-19T18:02:42Z]              at /rustc/09c42c45858d5f3aedfa670698275303a3d19afa/library/core/src/panicking.rs:101:14
  | [2021-11-19T18:02:42Z]    2: near_network::test_utils::WaitOrTimeoutActor::wait_or_timeout::{{closure}}
  | [2021-11-19T18:02:42Z]              at /var/lib/buildkite-agent/builds/buildkite-i-0cab78009f71c830c-1/nearprotocol/nearcore/chain/network/src/test_utils.rs:159:21
  | [2021-11-19T18:02:42Z]    3: <F as actix::utils::TimerFuncBox<A>>::call
  | [2021-11-19T18:02:42Z]              at /var/lib/buildkite-agent/.cargo/registry/src/github.com-1ecc6299db9ec823/actix-0.11.0-beta.2/src/utils.rs:117:9
  | [2021-11-19T18:02:42Z]    4: <actix::utils::TimerFunc<A> as actix::fut::ActorFuture>::poll
  | [2021-11-19T18:02:42Z]              at /var/lib/buildkite-agent/.cargo/registry/src/github.com-1ecc6299db9ec823/actix-0.11.0-beta.2/src/utils.rs:138:21
  | [2021-11-19T18:02:42Z]    5: <core::pin::Pin<P> as actix::fut::ActorFuture>::poll
  | [2021-11-19T18:02:42Z]              at /var/lib/buildkite-agent/.cargo/registry/src/github.com-1ecc6299db9ec823/actix-0.11.0-beta.2/src/fut/mod.rs:323:9
  | [2021-11-19T18:02:42Z]    6: <actix::contextimpl::ContextFut<A,C> as core::future::future::Future>::poll
  | [2021-11-19T18:02:42Z]              at /var/lib/buildkite-agent/.cargo/registry/src/github.com-1ecc6299db9ec823/actix-0.11.0-beta.2/src/contextimpl.rs:384:23
  | [2021-11-19T18:02:42Z]    7: tokio::runtime::task::core::CoreStage<T>::poll::{{closure}}
  | [2021-11-19T18:02:42Z]              at /var/lib/buildkite-agent/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.12.0/src/runtime/task/core.rs:161:17
  | [2021-11-19T18:02:42Z]    8: tokio::loom::std::unsafe_cell::UnsafeCell<T>::with_mut
  | [2021-11-19T18:02:42Z]              at /var/lib/buildkite-agent/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.12.0/src/loom/std/unsafe_cell.rs:14:9
  | [2021-11-19T18:02:42Z]    9: tokio::runtime::task::core::CoreStage<T>::poll
  | [2021-11-19T18:02:42Z]              at /var/lib/buildkite-agent/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.12.0/src/runtime/task/core.rs:151:13
  | [2021-11-19T18:02:42Z]   10: tokio::runtime::task::harness::poll_future::{{closure}}
  | [2021-11-19T18:02:42Z]              at /var/lib/buildkite-agent/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.12.0/src/runtime/task/harness.rs:461:19
  | [2021-11-19T18:02:42Z]   11: <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
  | [2021-11-19T18:02:42Z]              at /rustc/09c42c45858d5f3aedfa670698275303a3d19afa/library/core/src/panic/unwind_safe.rs:271:9
  | [2021-11-19T18:02:42Z]   12: std::panicking::try::do_call
  | [2021-11-19T18:02:42Z]              at /rustc/09c42c45858d5f3aedfa670698275303a3d19afa/library/std/src/panicking.rs:403:40
  | [2021-11-19T18:02:42Z]   13: __rust_try
  | [2021-11-19T18:02:42Z]   14: std::panicking::try
  | [2021-11-19T18:02:42Z]              at /rustc/09c42c45858d5f3aedfa670698275303a3d19afa/library/std/src/panicking.rs:367:19
  | [2021-11-19T18:02:42Z]   15: std::panic::catch_unwind
  | [2021-11-19T18:02:42Z]              at /rustc/09c42c45858d5f3aedfa670698275303a3d19afa/library/std/src/panic.rs:129:14
  | [2021-11-19T18:02:42Z]   16: tokio::runtime::task::harness::poll_future
  | [2021-11-19T18:02:42Z]              at /var/lib/buildkite-agent/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.12.0/src/runtime/task/harness.rs:449:18
  | [2021-11-19T18:02:42Z]   17: tokio::runtime::task::harness::Harness<T,S>::poll_inner
  | [2021-11-19T18:02:42Z]              at /var/lib/buildkite-agent/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.12.0/src/runtime/task/harness.rs:98:27
  | [2021-11-19T18:02:42Z]   18: tokio::runtime::task::harness::Harness<T,S>::poll
  | [2021-11-19T18:02:42Z]              at /var/lib/buildkite-agent/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.12.0/src/runtime/task/harness.rs:53:15
  | [2021-11-19T18:02:42Z]   19: tokio::runtime::task::raw::poll
  | [2021-11-19T18:02:42Z]              at /var/lib/buildkite-agent/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.12.0/src/runtime/task/raw.rs:113:5
  | [2021-11-19T18:02:42Z]   20: tokio::runtime::task::raw::RawTask::poll
  | [2021-11-19T18:02:42Z]              at /var/lib/buildkite-agent/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.12.0/src/runtime/task/raw.rs:70:18
  | [2021-11-19T18:02:42Z]   21: tokio::runtime::task::LocalNotified<S>::run
  | [2021-11-19T18:02:42Z]              at /var/lib/buildkite-agent/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.12.0/src/runtime/task/mod.rs:343:9
  | [2021-11-19T18:02:42Z]   22: tokio::task::local::LocalSet::tick::{{closure}}
  | [2021-11-19T18:02:42Z]              at /var/lib/buildkite-agent/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.12.0/src/task/local.rs:521:54
  | [2021-11-19T18:02:42Z]   23: tokio::coop::with_budget::{{closure}}
  | [2021-11-19T18:02:42Z]              at /var/lib/buildkite-agent/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.12.0/src/coop.rs:106:9
  | [2021-11-19T18:02:42Z]   24: std::thread::local::LocalKey<T>::try_with
  | [2021-11-19T18:02:42Z]              at /rustc/09c42c45858d5f3aedfa670698275303a3d19afa/library/std/src/thread/local.rs:399:16
  | [2021-11-19T18:02:42Z]   25: std::thread::local::LocalKey<T>::with
  | [2021-11-19T18:02:42Z]              at /rustc/09c42c45858d5f3aedfa670698275303a3d19afa/library/std/src/thread/local.rs:375:9
  | [2021-11-19T18:02:42Z]   26: tokio::coop::with_budget
  | [2021-11-19T18:02:42Z]              at /var/lib/buildkite-agent/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.12.0/src/coop.rs:99:5
  | [2021-11-19T18:02:42Z]   27: tokio::coop::budget
  | [2021-11-19T18:02:42Z]              at /var/lib/buildkite-agent/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.12.0/src/coop.rs:76:5
  | [2021-11-19T18:02:42Z]   28: tokio::task::local::LocalSet::tick
  | [2021-11-19T18:02:42Z]              at /var/lib/buildkite-agent/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.12.0/src/task/local.rs:521:31
  | [2021-11-19T18:02:42Z]   29: <tokio::task::local::RunUntil<T> as core::future::future::Future>::poll::{{closure}}
  | [2021-11-19T18:02:42Z]              at /var/lib/buildkite-agent/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.12.0/src/task/local.rs:646:16
  | [2021-11-19T18:02:42Z]   30: tokio::macros::scoped_tls::ScopedKey<T>::set
  | [2021-11-19T18:02:42Z]              at /var/lib/buildkite-agent/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.12.0/src/macros/scoped_tls.rs:61:9
  | [2021-11-19T18:02:42Z]   31: tokio::task::local::LocalSet::with
  | [2021-11-19T18:02:42Z]              at /var/lib/buildkite-agent/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.12.0/src/task/local.rs:559:9
  | [2021-11-19T18:02:42Z]   32: <tokio::task::local::RunUntil<T> as core::future::future::Future>::poll
  | [2021-11-19T18:02:42Z]              at /var/lib/buildkite-agent/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.12.0/src/task/local.rs:632:9
  | [2021-11-19T18:02:42Z]   33: tokio::task::local::LocalSet::run_until::{{closure}}
  | [2021-11-19T18:02:42Z]              at /var/lib/buildkite-agent/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.12.0/src/task/local.rs:505:9
  | [2021-11-19T18:02:42Z]   34: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
  | [2021-11-19T18:02:42Z]              at /rustc/09c42c45858d5f3aedfa670698275303a3d19afa/library/core/src/future/mod.rs:80:19
  | [2021-11-19T18:02:42Z]   35: <core::pin::Pin<P> as core::future::future::Future>::poll
  | [2021-11-19T18:02:42Z]              at /rustc/09c42c45858d5f3aedfa670698275303a3d19afa/library/core/src/future/future.rs:119:9
  | [2021-11-19T18:02:42Z]   36: tokio::runtime::basic_scheduler::Inner<P>::block_on::{{closure}}::{{closure}}
  | [2021-11-19T18:02:42Z]              at /var/lib/buildkite-agent/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.12.0/src/runtime/basic_scheduler.rs:230:62
  | [2021-11-19T18:02:42Z]   37: tokio::coop::with_budget::{{closure}}
  | [2021-11-19T18:02:42Z]              at /var/lib/buildkite-agent/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.12.0/src/coop.rs:106:9
  | [2021-11-19T18:02:42Z]   38: std::thread::local::LocalKey<T>::try_with
  | [2021-11-19T18:02:42Z]              at /rustc/09c42c45858d5f3aedfa670698275303a3d19afa/library/std/src/thread/local.rs:399:16
  | [2021-11-19T18:02:42Z]   39: std::thread::local::LocalKey<T>::with
  | [2021-11-19T18:02:42Z]              at /rustc/09c42c45858d5f3aedfa670698275303a3d19afa/library/std/src/thread/local.rs:375:9
  | [2021-11-19T18:02:42Z]   40: tokio::coop::with_budget
  | [2021-11-19T18:02:42Z]              at /var/lib/buildkite-agent/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.12.0/src/coop.rs:99:5
  | [2021-11-19T18:02:42Z]   41: tokio::coop::budget
  | [2021-11-19T18:02:42Z]              at /var/lib/buildkite-agent/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.12.0/src/coop.rs:76:5
  | [2021-11-19T18:02:42Z]   42: tokio::runtime::basic_scheduler::Inner<P>::block_on::{{closure}}
  | [2021-11-19T18:02:42Z]              at /var/lib/buildkite-agent/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.12.0/src/runtime/basic_scheduler.rs:230:39
  | [2021-11-19T18:02:42Z]   43: tokio::runtime::basic_scheduler::enter::{{closure}}
  | [2021-11-19T18:02:42Z]              at /var/lib/buildkite-agent/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.12.0/src/runtime/basic_scheduler.rs:339:29
  | [2021-11-19T18:02:42Z]   44: tokio::macros::scoped_tls::ScopedKey<T>::set
  | [2021-11-19T18:02:42Z]              at /var/lib/buildkite-agent/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.12.0/src/macros/scoped_tls.rs:61:9
  | [2021-11-19T18:02:42Z]   45: tokio::runtime::basic_scheduler::enter
  | [2021-11-19T18:02:42Z]              at /var/lib/buildkite-agent/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.12.0/src/runtime/basic_scheduler.rs:339:5
  | [2021-11-19T18:02:42Z]   46: tokio::runtime::basic_scheduler::Inner<P>::block_on
  | [2021-11-19T18:02:42Z]              at /var/lib/buildkite-agent/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.12.0/src/runtime/basic_scheduler.rs:218:9
  | [2021-11-19T18:02:42Z]   47: tokio::runtime::basic_scheduler::InnerGuard<P>::block_on
  | [2021-11-19T18:02:42Z]              at /var/lib/buildkite-agent/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.12.0/src/runtime/basic_scheduler.rs:489:9
  | [2021-11-19T18:02:42Z]   48: tokio::runtime::basic_scheduler::BasicScheduler<P>::block_on
  | [2021-11-19T18:02:42Z]              at /var/lib/buildkite-agent/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.12.0/src/runtime/basic_scheduler.rs:178:24
  | [2021-11-19T18:02:42Z]   49: tokio::runtime::Runtime::block_on
  | [2021-11-19T18:02:42Z]              at /var/lib/buildkite-agent/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.12.0/src/runtime/mod.rs:461:46
  | [2021-11-19T18:02:42Z]   50: tokio::task::local::LocalSet::block_on
  | [2021-11-19T18:02:42Z]              at /var/lib/buildkite-agent/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.12.0/src/task/local.rs:466:9
  | [2021-11-19T18:02:42Z]   51: actix_rt::runtime::Runtime::block_on
  | [2021-11-19T18:02:42Z]              at /var/lib/buildkite-agent/.cargo/registry/src/github.com-1ecc6299db9ec823/actix-rt-2.3.0/src/runtime.rs:80:9
  | [2021-11-19T18:02:42Z]   52: actix_rt::system::SystemRunner::run
  | [2021-11-19T18:02:42Z]              at /var/lib/buildkite-agent/.cargo/registry/src/github.com-1ecc6299db9ec823/actix-rt-2.3.0/src/system.rs:167:15
  | [2021-11-19T18:02:42Z]   53: near_actix_test_utils::run_actix
  | [2021-11-19T18:02:42Z]              at /var/lib/buildkite-agent/builds/buildkite-i-0cab78009f71c830c-1/nearprotocol/nearcore/test-utils/actix-test-utils/src/lib.rs:103:5
  | [2021-11-19T18:02:42Z]   54: network::runner::start_test
  | [2021-11-19T18:02:42Z]              at ./tests/network/runner.rs:680:5
  | [2021-11-19T18:02:42Z]   55: network::routing::archival_node
  | [2021-11-19T18:02:42Z]              at ./tests/network/routing.rs:284:5
  | [2021-11-19T18:02:42Z]   56: network::routing::archival_node::{{closure}}
  | [2021-11-19T18:02:42Z]              at ./tests/network/routing.rs:259:1
  | [2021-11-19T18:02:42Z]   57: core::ops::function::FnOnce::call_once
  | [2021-11-19T18:02:42Z]              at /rustc/09c42c45858d5f3aedfa670698275303a3d19afa/library/core/src/ops/function.rs:227:5
  | [2021-11-19T18:02:42Z]   58: core::ops::function::FnOnce::call_once
  | [2021-11-19T18:02:42Z]              at /rustc/09c42c45858d5f3aedfa670698275303a3d19afa/library/core/src/ops/function.rs:227:5
  | [2021-11-19T18:02:42Z] note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
  | [2021-11-19T18:02:42Z] thread 'routing::archival_node' panicked at 'called `Result::unwrap()` on an `Err` value: Custom { kind: Other, error: "Non-zero exit code: 1" }', /var/lib/buildkite-agent/builds/buildkite-i-0cab78009f71c830c-1/nearprotocol/nearcore/test-utils/actix-test-utils/src/lib.rs:103:15
  | [2021-11-19T18:02:42Z] stack backtrace:
  | [2021-11-19T18:02:42Z]    0: rust_begin_unwind
  | [2021-11-19T18:02:42Z]              at /rustc/09c42c45858d5f3aedfa670698275303a3d19afa/library/std/src/panicking.rs:517:5
  | [2021-11-19T18:02:42Z]    1: core::panicking::panic_fmt
  | [2021-11-19T18:02:42Z]              at /rustc/09c42c45858d5f3aedfa670698275303a3d19afa/library/core/src/panicking.rs:101:14
  | [2021-11-19T18:02:42Z]    2: core::result::unwrap_failed
  | [2021-11-19T18:02:42Z]              at /rustc/09c42c45858d5f3aedfa670698275303a3d19afa/library/core/src/result.rs:1617:5
  | [2021-11-19T18:02:42Z]    3: core::result::Result<T,E>::unwrap
  | [2021-11-19T18:02:42Z]              at /rustc/09c42c45858d5f3aedfa670698275303a3d19afa/library/core/src/result.rs:1299:23
  | [2021-11-19T18:02:42Z]    4: near_actix_test_utils::run_actix
  | [2021-11-19T18:02:42Z]              at /var/lib/buildkite-agent/builds/buildkite-i-0cab78009f71c830c-1/nearprotocol/nearcore/test-utils/actix-test-utils/src/lib.rs:103:5
  | [2021-11-19T18:02:42Z]    5: network::runner::start_test
  | [2021-11-19T18:02:42Z]              at ./tests/network/runner.rs:680:5
  | [2021-11-19T18:02:42Z]    6: network::routing::archival_node
  | [2021-11-19T18:02:42Z]              at ./tests/network/routing.rs:284:5
  | [2021-11-19T18:02:42Z]    7: network::routing::archival_node::{{closure}}
  | [2021-11-19T18:02:42Z]              at ./tests/network/routing.rs:259:1
  | [2021-11-19T18:02:42Z]    8: core::ops::function::FnOnce::call_once
  | [2021-11-19T18:02:42Z]              at /rustc/09c42c45858d5f3aedfa670698275303a3d19afa/library/core/src/ops/function.rs:227:5
  | [2021-11-19T18:02:42Z]    9: core::ops::function::FnOnce::call_once
  | [2021-11-19T18:02:42Z]              at /rustc/09c42c45858d5f3aedfa670698275303a3d19afa/library/core/src/ops/function.rs:227:5
  | [2021-11-19T18:02:42Z] note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
  | [2021-11-19T18:02:42Z]
  | [2021-11-19T18:02:42Z]
  | [2021-11-19T18:02:42Z] failures:
  | [2021-11-19T18:02:42Z]     routing::archival_node
  | [2021-11-19T18:02:42Z]
  | [2021-11-19T18:02:42Z] test result: FAILED. 28 passed; 1 failed; 1 ignored; 0 measured; 0 filtered out; finished in 34.32s


```

**Comment by pmnoxx** [comment]
I'll add debug logs to `runner.rs` first. This will make it possible to figure out what's going on based on logs:
https://github.com/near/nearcore/pull/5390/files

**Comment by pmnoxx** [comment]
Let's ignore this test for now. I'll work on it next week, but I don't want people from merging.
https://github.com/near/nearcore/pull/5399

**Comment by pmnoxx** [comment]
The test broke due to
```
commit 07dadeac0a4db8a1ff448fcf1174e706badf54eb
Author: pmnoxx <piotr.mikulski@gmail.com>
Date:   Wed Nov 17 08:40:46 2021 -0800

    bug: fix `network/routing.rs` tests (#5325)
    
    In `PeerActor` we have a logic which is used to drop duplicated messages if they happen within `50`ms. That constant used to be higher while the change was originally implemented.
    ```
    /// Duplicated messages will be dropped if routed through the same peer multiple times.
    pub const DROP_DUPLICATED_MESSAGES_PERIOD: Duration = Duration::from_millis(50);
    ```
    
    In `network/routing.rs` tests we use structs `StateMachine` to test PR. The check coincidentally runs every `50ms`, which is sometimes too fast, sometimes too slow for the logic to drop duplicated messages to work.
    
    The fix is to change constant inside testes from `50ms` to `1ms`.
    
    Closes #5322
```
